<a href="https://colab.research.google.com/github/Vanitharavi245/PRODIGY_TrackCode_TaskNumber-1/blob/main/Model_1_for_internship.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Import necessary libraries

In [ ]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

#Importing dataset

In [ ]:
train_data=pd.read_csv("/content/train (1).csv")
test_data=pd.read_csv("/content/test (1).csv")

In [ ]:
column_names_index = train_data.columns

# To convert the column names to a list
column_names_list = train_data.columns.tolist()

# Display column names
print("Column Names (Index object):", column_names_index)
print("Column Names (List):", column_names_list)

Column Names (Index object): Index(['Id', 'MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea', 'Street',
       'Alley', 'LotShape', 'LandContour', 'Utilities', 'LotConfig',
       'LandSlope', 'Neighborhood', 'Condition1', 'Condition2', 'BldgType',
       'HouseStyle', 'OverallQual', 'OverallCond', 'YearBuilt', 'YearRemodAdd',
       'RoofStyle', 'RoofMatl', 'Exterior1st', 'Exterior2nd', 'MasVnrType',
       'MasVnrArea', 'ExterQual', 'ExterCond', 'Foundation', 'BsmtQual',
       'BsmtCond', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinSF1',
       'BsmtFinType2', 'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', 'Heating',
       'HeatingQC', 'CentralAir', 'Electrical', '1stFlrSF', '2ndFlrSF',
       'LowQualFinSF', 'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath',
       'HalfBath', 'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual',
       'TotRmsAbvGrd', 'Functional', 'Fireplaces', 'FireplaceQu', 'GarageType',
       'GarageYrBlt', 'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual',
   

#Selecting Respective Features

In [ ]:
X_train=train_data[['GrLivArea', 'BedroomAbvGr', 'FullBath']]
y_train = train_data['SalePrice']

#Selecting features for test data

In [ ]:
X_test=test_data[['GrLivArea', 'BedroomAbvGr', 'FullBath']]


#Train the Linear Regression model

In [ ]:
# Train the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

#Prediction on Test Data

In [ ]:
# Predict house prices on the test dataset
y_pred = model.predict(X_test)


#Display the prediction value

In [ ]:
# Display predicted values
print("Predicted House Prices:", y_pred)

Predicted House Prices: [120100.81297664 139898.20827909 202611.4145861  ... 100482.39478483
 100386.04850582 243444.31507603]


#Cross Validation

In [ ]:
from sklearn.model_selection import cross_val_score
import numpy as np

# Perform cross-validation
cv_scores = cross_val_score(model, X_train, y_train, cv=5, scoring='neg_mean_squared_error')
cv_rmse = np.sqrt(-cv_scores.mean())
print(f"Cross-Validated RMSE: {cv_rmse}")


Cross-Validated RMSE: 51764.286681239115


In [ ]:
# Calculate the baseline RMSE (predicting the mean of y_train)
baseline_pred = np.mean(y_train)
baseline_rmse = np.sqrt(mean_squared_error(y_train, [baseline_pred] * len(y_train)))

print(f"Baseline RMSE: {baseline_rmse}")
print(f"Model RMSE: {cv_rmse}")

if cv_rmse < baseline_rmse:
    print("Model performs better than the baseline.")
else:
    print("Model does not perform better than the baseline.")


Baseline RMSE: 79415.29188606751
Model RMSE: 51764.286681239115
Model performs better than the baseline.


##Interpretation about model performance

My model’s RMSE of 51,764.29 is significantly lower than the baseline RMSE of 79,415.29, indicating that your model is performing better than simply predicting the mean house price. This suggests that my  model is capturing patterns in the data and providing more accurate predictions compared to a basic approach.